app/channel.go

```diff
@@ -411,12 +451,49 @@ func (a *App) createGroupChannel(userIds []string, creatorId string) (*model.Cha
 		Type:        model.CHANNEL_GROUP,
 	}
 
-	channel, err := a.Srv().Store.Channel().Save(group, *a.Config().TeamSettings.MaxChannelsPerTeam)
-	if err != nil {
-		if err.Id == store.CHANNEL_EXISTS_ERROR {
-			return channel, err
+	channel, nErr := a.Srv().Store.Channel().Save(group, *a.Config().TeamSettings.MaxChannelsPerTeam)
+	if nErr != nil {
+		var invErr *store.ErrInvalidInput
+		var intErr *store.ErrInternal
+		var cErr *store.ErrConflict
+		var ltErr *store.ErrLimitExceeded
+		var appErr *model.AppError
+		switch {
+		case errors.As(nErr, &invErr):
+			switch {
+			case invErr.Entity == "Channel" && invErr.Field == "DeleteAt":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.archived_channel.app_error", nil, "", http.StatusBadRequest)
+			case invErr.Entity == "Channel" && invErr.Field == "Type":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.direct_channel.app_error", nil, "", http.StatusBadRequest)
+			case invErr.Entity == "Channel" && invErr.Field == "Id":
+				return nil, model.NewAppError("SqlChannelStore.Save", "store.sql_channel.save_channel.existing.app_error", nil, "id="+invErr.Value.(string), http.StatusBadRequest)
```

Comment:
Probably we can set all the wheres here as `SqlChannelStore.Save` not only this one, because the errors is returned from there.

Reply:
Hmm .. actually I made a mistake. IMO we should be setting the `where` as the origin point from where the AppError was instantiated. And the AppError was instantiated from `CreateChannel`. That itself contains another error internally which contains detailed info. But we should not set `where` as the origin point of the embedded error. Let me know what you think.

Reply:
Yes, lets keep it as `CreateChannel` because is where is generated, and is like that for the rest of the source code, and I don't want to include more changes in this PR. I think what you proposed is the most consistent approach with the rest of the code.